leo_docking/leo_docking/docking_server.py

```diff
@@ -190,8 +41,68 @@ def __init__(self):
         )
 
         # Create and start the introspection server
-        self._smach_introspection_server = smach_ros.IntrospectionServer(
-            "leo_docking", self._state_machine, "/LEO_DOCKING"
+        self._smach_introspection_server = IntrospectionServer(
+            "leo_docking", self._state_machine.state_machine, "/LEO_DOCKING"
+        )
+
+    def _init_ros(self):
+        sub_qos = QoSProfile(
+            reliability=QoSReliabilityPolicy.BEST_EFFORT, durability=QoSDurabilityPolicy.VOLATILE, depth=1
+        )
+        self.create_subscription(ArucoDetection, "/aruco_detections", self._aruco_detection_cb, qos_profile=sub_qos)
+        self.create_subscription(Odometry, "/odometry/filtered/local", self._wheel_odom_cb, qos_profile=sub_qos)
```

Comment:
Could be nice to set the default odom topic for Leo Rover here. I do not right now remember whether it is /odom or /odometry/filtered or even based on the name of the callback it could be the /wheel_"something"_"something" topic... 😅

The /odometry/filtered//local is definitely our internal topic name